MarkBind/markbind
# Upgrade vulnerable MarkBind dependencies

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Other, please explain: upgrading dependencies


Part of #461.



**What is the rationale for this request?**
[`npm audit`](https://docs.npmjs.com/auditing-package-dependencies-for-security-vulnerabilities) (avaliable from `npm >= 6.0.0`) returns **434** vulnerabilities in our dependencies. We should upgrade vulnerable packages to remove these vulnerabilities.

[Full results here.](https://gist.github.com/Xenonym/ea149da86bd83307e42c3503eba66b4c#file-audit-txt)

**What changes did you make? (Give an overview)**
- I upgraded the following vulnerable packages, categoried by their function in MarkBind:
	- Live preview
		- `chokidar`: 1.6.1 → 2.1.5
		- `live-server`: 1.2.0 → 1.2.1
	- GitHub Pages deployment
		- `gh-pages`: 1.1.0 → 2.0.1
	- Building pages
		- `fastmatter`: 2.0.1 → 2.1.0
		- `nunjucks`: 3.0.0 → 3.2.0
		- `markdown-it-table-of-contents`: 0.3.2 → 0.4.3
	- Development tools
		- `eslint`: 4.16.0 → 5.15.3
		- `jest`: 22.4.3 → 24.5.0
		- `lodash`: 4.17.5 → 4.17.11
- I also upgraded the `eslint` config packages, as there are rule specifications that have changed/deprecated between the `eslint` versions:
	- `eslint-config-airbnb-base`: 12.1.0 → 13.1.0
	- `eslint-plugin-import`: 2.8.0 → 2.16.0
	- `eslint-plugin-lodash`: 22.4.3 → 24.5.0
- I updated `.eslintrc.js` to accommodate our current coding style and disable `no-else-return` and `implicit-arrow-linebreak` as we do not currently follow these rules.
- I changed the import order in some JS files to match the `sort-import` rule (package dependencies before local dependencies).

**Is there anything you'd like reviewers to focus on?**
- Is the current `.eslintrc.js` suitable now that we have updated our configs?

**Testing instructions:**
1. Ensure that `npm audit` returns 0 vulnerabilities.
1. Ensure that existing functionality under the four areas affected (live preview, GitHub Pages deployment, page building, other dev tools) function as expected.
